### ISSUE TITLE
[Bug] gemma-7b-it-q4f16_2-MLC  crashes on SLM flow cuda12.2

### ISSUE THREAD

**Issue by Sing-Li** [mlc-ai/mlc-llm#1913]
## 🐛 Bug

When running python chat with gemma-7b-it-q4f16_2-MLC ,  system crashes with:

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 1, column: 1)', src/lib.rs:21:50
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
```

## To Reproduce

On a known good SLM flow cuda12.2 system, run the gemma-7b model...
```
$ python3 -m mlc_chat chat HF://mlc-ai/gemma-7b-it-q4f16_2-MLC
```

System will crash with....
```
[2024-03-09 02:46:18] INFO model_metadata.py:96: Total memory usage: 5856.34 MB (Parameters: 5658.33 MB. KVCache: 0.00 MB. Temporary buffer: 198.01 MB)
[2024-03-09 02:46:18] INFO model_metadata.py:105: To reduce memory usage, tweak `prefill_chunk_size`, `context_window_size` and `sliding_window_size`
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 1, column: 1)', src/lib.rs:21:50
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
Aborted (core dumped)
```

## Expected behavior

Chat to work with the gemma-7b model.

## Environment

 - Platform (e.g. WebGPU/Vulkan/IOS/Android/CUDA):  CUDA 12.2
 - Operating system (e.g. Ubuntu/Windows/MacOS/...): Ubuntu 22.04lts
 - Device (e.g. iPhone 12 Pro, PC+RTX 3090, ...)  3060 12GB
 - How you installed MLC-LLM (`conda`, source):  conda
 - How you installed TVM-Unity (`pip`, source):  pip
 - Python version (e.g. 3.10):  3.10
 - GPU driver version (if applicable):
 - CUDA/cuDNN version (if applicable):
 - TVM Unity Hash Tag (`python -c "import tvm; print('\n'.join(f'{k}: {v}' for k, v in tvm.support.libinfo().items()))"`, applicable if you compile models):
 - Any other relevant information:

## Additional context

Actual run with trace ...

```
python3 -m mlc_chat chat HF://mlc-ai/gemma-7b-it-q4f16_2-MLC
[2024-03-09 02:44:46] INFO auto_device.py:76: Found device: cuda:0
[2024-03-09 02:44:47] INFO auto_device.py:85: Not found device: rocm:0
[2024-03-09 02:44:47] INFO auto_device.py:85: Not found device: metal:0
[2024-03-09 02:44:48] INFO auto_device.py:76: Found device: vulkan:0
[2024-03-09 02:44:49] INFO auto_device.py:85: Not found device: opencl:0
[2024-03-09 02:44:49] INFO auto_device.py:33: Using device: cuda:0
[2024-03-09 02:44:49] INFO chat_module.py:356: Downloading model from HuggingFace: HF://mlc-ai/gemma-7b-it-q4f16_2-MLC
[2024-03-09 02:44:49] INFO download.py:39: [Git] Cloning https://huggingface.co/mlc-ai/gemma-7b-it-q4f16_2-MLC.git to /tmp/tmpl5xfv6qo/tmp
[2024-03-09 02:44:51] INFO download.py:144: Downloaded https://huggingface.co/mlc-ai/gemma-7b-it-q4f16_2-MLC/resolve/main/params_shard_1.bin to /tmp/tmpl5xfv6qo/tmp/params_shard_1.bin
[2024-03-09 02:44:51] INFO download.py:144: Downloaded https://huggingface.co/mlc-ai/gemma-7b-it-q4f16_2-MLC/resolve/main/params_shard_3.bin to /tmp/tmpl5xfv6qo/tmp/params_shard_3.bin
[2024-03-09 02:44:52] INFO download.py:144: Downloaded https://huggingface.co/mlc-ai/gemma-7b-it-q4f16_2-...

[2024-03-09 02:45:45] INFO download.py:144: Downloaded https://huggingface.co/mlc-ai/gemma-7b-it-q4f16_2-MLC/resolve/main/params_shard_99.bin to /tmp/tmpl5xfv6qo/tmp/params_shard_99.bin
100%|███████████████████████████████████████████████████████████████████████████████████████████| 101/101 [00:55<00:00,  1.83it/s]
[2024-03-09 02:45:45] INFO download.py:145: Moving /tmp/tmpl5xfv6qo/tmp to /home/autoqa/.cache/mlc_chat/model_weights/mlc-ai/gemma-7b-it-q4f16_2-MLC
[2024-03-09 02:45:45] INFO chat_module.py:765: Model lib not found. Now compiling model lib on device...
[2024-03-09 02:45:45] INFO jit.py:34: MLC_JIT_POLICY = ON. Can be one of: ON, OFF, REDO, READONLY
[2024-03-09 02:45:45] INFO jit.py:93: Compiling using commands below:
[2024-03-09 02:45:45] INFO jit.py:94: /usr/bin/python3 -m mlc_chat compile /home/autoqa/.cache/mlc_chat/model_weights/mlc-ai/gemma-7b-it-q4f16_2-MLC --opt 'flashinfer=1;cublas_gemm=1;faster_transformer=1;cudagraph=0' --overrides 'context_window_size=8192;prefill_chunk_size=1024;tensor_parallel_shards=1' --device cuda:0 --output /tmp/tmploqlcscr/lib.so
[2024-03-09 02:45:45] INFO auto_config.py:69: Found model configuration: /home/autoqa/.cache/mlc_chat/model_weights/mlc-ai/gemma-7b-it-q4f16_2-MLC/mlc-chat-config.json
[2024-03-09 02:45:45] INFO auto_target.py:83: Detecting target device: cuda:0
[2024-03-09 02:45:45] INFO auto_target.py:85: Found target: {"thread_warp_size": 32, "arch": "sm_86", "max_threads_per_block": 1024, "max_num_threads": 1024, "kind": "cuda", "max_shared_memory_per_block": 49152, "tag": "", "keys": ["cuda", "gpu"]}
[2024-03-09 02:45:45] INFO auto_target.py:102: Found host LLVM triple: x86_64-redhat-linux-gnu
[2024-03-09 02:45:45] INFO auto_target.py:103: Found host LLVM CPU: znver3
[2024-03-09 02:45:45] INFO auto_target.py:269: Generating code for CUDA architecture: sm_86
[2024-03-09 02:45:45] INFO auto_target.py:270: To produce multi-arch fatbin, set environment variable MLC_MULTI_ARCH. Example: MLC_MULTI_ARCH=70,72,75,80,86,87,89,90
[2024-03-09 02:45:45] INFO auto_config.py:153: Found model type: gemma. Use `--model-type` to override.
Compiling with arguments:
  --config          GemmaConfig(hidden_size=3072, hidden_act='gelu', intermediate_size=24576, attention_bias=False, num_attention_heads=16, num_key_value_heads=16, head_dim=256, num_hidden_layers=28, rms_norm_eps=1e-06, vocab_size=256000, position_embedding_base=10000.0, context_window_size=8192, prefill_chunk_size=1024, tensor_parallel_shards=1, max_batch_size=80, kwargs={})
  --quantization    GroupQuantize(name='q4f16_2', kind='group-quant', group_size=32, quantize_dtype='int4', storage_dtype='uint32', model_dtype='float16', linear_weight_layout='NK', quantize_embedding=False, quantize_final_fc=False, num_elem_per_storage=8, num_storage_per_group=4, max_int_value=7)
  --model-type      gemma
  --target          {"thread_warp_size": 32, "host": {"mtriple": "x86_64-redhat-linux-gnu", "tag": "", "kind": "llvm", "mcpu": "znver3", "keys": ["cpu"]}, "arch": "sm_86", "max_threads_per_block": 1024, "libs": ["thrust"], "max_num_threads": 1024, "kind": "cuda", "max_shared_memory_per_block": 49152, "tag": "", "keys": ["cuda", "gpu"]}
  --opt             flashinfer=1;cublas_gemm=0;faster_transformer=1;cudagraph=0
  --system-lib-prefix ""
  --output          /tmp/tmploqlcscr/lib.so
  --overrides       context_window_size=8192;sliding_window_size=None;prefill_chunk_size=1024;attention_sink_size=None;max_batch_size=None;tensor_parallel_shards=1
[2024-03-09 02:45:45] INFO config.py:106: Overriding context_window_size from 8192 to 8192
[2024-03-09 02:45:45] INFO config.py:106: Overriding prefill_chunk_size from 1024 to 1024
[2024-03-09 02:45:45] INFO config.py:106: Overriding tensor_parallel_shards from 1 to 1
[2024-03-09 02:45:45] INFO compile.py:136: Creating model from: GemmaConfig(hidden_size=3072, hidden_act='gelu', intermediate_size=24576, attention_bias=False, num_attention_heads=16, num_key_value_heads=16, head_dim=256, num_hidden_layers=28, rms_norm_eps=1e-06, vocab_size=256000, position_embedding_base=10000.0, context_window_size=8192, prefill_chunk_size=1024, tensor_parallel_shards=1, max_batch_size=80, kwargs={})
[2024-03-09 02:45:45] INFO compile.py:155: Exporting the model to TVM Unity compiler
[2024-03-09 02:45:46] INFO compile.py:161: Running optimizations using TVM Unity
[2024-03-09 02:45:46] INFO compile.py:174: Registering metadata: {'model_type': 'gemma', 'quantization': 'q4f16_2', 'context_window_size': 8192, 'sliding_window_size': -1, 'attention_sink_size': -1, 'prefill_chunk_size': 1024, 'tensor_parallel_shards': 1, 'kv_cache_bytes': 0}
[2024-03-09 02:45:46] INFO pipeline.py:46: Running TVM Relax graph-level optimizations
[2024-03-09 02:46:00] INFO pipeline.py:46: Lowering to TVM TIR kernels
[2024-03-09 02:46:01] INFO pipeline.py:46: Running TVM TIR-level optimizations
[2024-03-09 02:46:04] INFO pipeline.py:46: Running TVM Dlight low-level optimizations
[2024-03-09 02:46:10] INFO pipeline.py:46: Lowering to VM bytecode
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `alloc_embedding_tensor`: 6.00 MB
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `batch_decode`: 15.47 MB
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `batch_prefill`: 198.00 MB
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `batch_verify`: 198.00 MB
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `create_tir_paged_kv_cache`: 0.00 MB
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `decode`: 0.19 MB
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `embed`: 12.00 MB
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `prefill`: 198.01 MB
[2024-03-09 02:46:11] INFO estimate_memory_usage.py:55: [Memory usage] Function `softmax_with_temperature`: 0.00 MB
[2024-03-09 02:46:11] INFO pipeline.py:46: Compiling external modules
[2024-03-09 02:46:11] INFO pipeline.py:46: Compilation complete! Exporting to disk
[2024-03-09 02:46:18] INFO model_metadata.py:96: Total memory usage: 5856.34 MB (Parameters: 5658.33 MB. KVCache: 0.00 MB. Temporary buffer: 198.01 MB)
[2024-03-09 02:46:18] INFO model_metadata.py:105: To reduce memory usage, tweak `prefill_chunk_size`, `context_window_size` and `sliding_window_size`
[2024-03-09 02:46:18] INFO compile.py:194: Generated: /tmp/tmploqlcscr/lib.so
[2024-03-09 02:46:18] INFO jit.py:97: Using compiled model lib: /home/autoqa/.cache/mlc_chat/model_lib/45db1fcb4a89e73226a62efc86ae4174.so
[2024-03-09 02:46:18] INFO model_metadata.py:96: Total memory usage: 5856.34 MB (Parameters: 5658.33 MB. KVCache: 0.00 MB. Temporary buffer: 198.01 MB)
[2024-03-09 02:46:18] INFO model_metadata.py:105: To reduce memory usage, tweak `prefill_chunk_size`, `context_window_size` and `sliding_window_size`
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 1, column: 1)', src/lib.rs:21:50
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
Aborted (core dumped)
```

**Comment by shreygupta2809** [comment]
Thanks, I am able to reproduce this, will look into it.

**Comment by MasterJH5574** [comment]
Hi @Sing-Li, we fixed it in #1954. It turns out that the issue is because the `tokenizer.json` file is a Git LFS file, and is not downloaded to your local machine, and thus errors when loading the tokenizer.
